### PR TITLE
[4415] Components Panel DnD for creating new component fails when content type has textarea or boolean fields

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4415

### CODE REVIEW

File: ui/guest/src/utils/contentType.ts [hunk 1]
```diff
@@ -75,9 +76,20 @@ export function getDefaultValue(field: ContentTypeField): string {
         return 'https://via.placeholder.com/150';
```
Comment (rart) [inline ui/guest/src/utils/contentType.ts]: Try using image dimensions from content type definition to set the right pic size

**Review by rart — COMMENTED** [review]
Is this finished?

File: ui/guest/src/utils/contentType.ts [hunk 2]
```diff
@@ -71,13 +72,30 @@ export function getDefaultValue(field: ContentTypeField): string {
...
-        // TODO use validation width, height
-        return 'https://via.placeholder.com/150';
+        // since it's a string, empty string needs to be validated
+        const width = JSON.parse(field.validations.width?.value).min
+          ? JSON.parse(field.validations.width?.value).min
+          : 150;
+        const height = JSON.parse(field.validations.height?.value).min
+          ? JSON.parse(field.validations.height?.value).min
+          : width;
+        return `https://via.placeholder.com/${width}x${height}`;
```
Comment (rart) [inline ui/guest/src/utils/contentType.ts]: What's a string?
By this time should be whatever it must be. Parsing should go to the service.

File: ui/app/src/services/contentTypes.ts [hunk 3]
```diff
@@ -40,15 +40,17 @@ const typeMap = {
...
 const systemValidationsKeysMap = {
   minSize: 'minCount',
   maxSize: 'maxCount',
   maxlength: 'maxLength',
   contentTypes: 'allowedContentTypes',
   tags: 'allowedContentTypeTags',
-  readonly: 'readOnly'
+  readonly: 'readOnly',
+  width: 'width',
+  height: 'height'
```
Comment (rart) [inline ui/app/src/services/contentTypes.ts]: Do we currently have min/max width/height? Do we have both exact width/height & max/min or just one?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/utils/contentType.ts (hunk 2 above)
Comment (jvega190) [reply ui/guest/src/utils/contentType.ts]: Done parsing in the service. Also, new fields for minWidth, maxWidth, minHeight and maxHeight

File: ui/app/src/services/contentTypes.ts (hunk 3 above)
Comment (jvega190) [reply ui/app/src/services/contentTypes.ts]: Added new fields for minWidth, maxWidth, minHeight and maxHeight

File: ui/app/src/services/contentTypes.ts [hunk 4]
```diff
@@ -71,7 +89,29 @@ function getFieldValidations(
...
+            name: prop.name,
+            type: prop.type,
+            value: parsedValidation.min
+          };
+          table[`max${capitalize(prop.name)}`] = {
+            name: prop.name,
+            type: prop.type,
+            value: parsedValidation.max
+          };
+        }
```
Comment (jvega190) [inline ui/app/src/services/contentTypes.ts:111]: @rart just for you to know, values are set unparsed in here, but later in code values are parsed before returning the validation fields.

Comment (rart) [reply ui/app/src/services/contentTypes.ts:111]: - Where about?
- Why wait/distribute?

File: ui/guest/src/classes/ICERegistry.ts [hunk 5]
```diff
@@ -59,7 +59,13 @@ const validationChecks: { [key in ValidationKeys]: Function } = {
...
   maxLength() {},
   readOnly() {},
-  required() {}
+  required() {},
+  width() {},
+  height() {},
+  minWidth() {},
+  minHeight() {},
+  maxWidth() {},
+  maxHeight() {}
```
Comment (rart) [inline ui/guest/src/classes/ICERegistry.ts:68]: What were the purpose of these? How come we have so many dummy functions? @aachavarria

Comment (aachavarria) [reply ui/guest/src/classes/ICERegistry.ts:68]: That functions are used when the `drop_zone_enter` and `drop_zone_leave` events are triggered.
Right now we are only checking validation for "minCount" and "maxCount".
There are a pending ticket to add more validations: https://github.com/craftercms/craftercms/issues/4052

**Review by rart — APPROVED** [review]

**Pull request merged**